server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterStatsAction.java

```diff
@@ -68,6 +68,8 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         clusterStatsRequest.timeout(request.param("timeout"));
         clusterStatsRequest.setIncludeDiscoveryNodes(false);
         clusterStatsRequest.useAggregatedNodeLevelResponses(true);
+        clusterStatsRequest.setIncludeMappingStats(request.paramAsBoolean("include_mapping_stats", true));
+        clusterStatsRequest.setIncludeAnalysisStats(request.paramAsBoolean("include_analysis_stats", true));
```

Comment:
APIs like Index Stats, Node Stats already provide a way to handle filtering the required stats through URI Path. Is it possible to implement a similar logic here?

e.g. index stats `/{index}/_stats/{metric}`
e.g. node stats `/_nodes/{nodeId}/stats/{metric}/{index_metric}`

We can add a similar route like `/_cluster/stats/{metrics}/{sub_metrics}` where `metrics` can be `nodes` or `indices` or both. And sub metrics can be specific details within stats like `mappings`, `analysis`  

This should help maintain a common pattern across different stats APIS

Reply:
Thanks for the inputs @mgodwan but the use-case here is little different. In the the URI path approach client can fetch stats for a specific metric whereas for this issue we want to add support for **filtering out** some information from the overall response that client does not need.

Reply:
> we want to add support for filtering out some information from the overall response that client does not need.

The motivation behind this PR seems to be https://github.com/opensearch-project/OpenSearch/pull/14447 which mentions `Reduced Overhead` and `Fine-Grained Control` as the benefits

Hence, opting-in to fetch required metrics still seem to be a way which we should explore to handle as it aligns with these benefits while keeping the API signature streamlined across different stats APIs. 

e.g. if new fields are added to the cluster stats response in a new OS version, users may start to see the overhead similar to what is mentioned in the parent issue due to some new computation happening. Opt-in mechanism like URI path ensures users only get what they want from cluster stats if they are sensitive to performance of these APIs or are concerned about computation overhead of these on the cluster. And as existing APIs work, if no filter is provided, it defaults to ALL which means all stats should be returned, and continues to maintain a backward compatible way for this existing API as well.

Reply:
I get the concerns, but I don't understand how we can use the URI paths to selective exclude some stats from the default response. For APIs like ```_cluster/health``` we support request params like ```include_defaults``` to additionally fetch default settings in addition to the default response. Conversely, in this use-case request params is being used to exclude some information from the default response.

By default the value of the params is set to true which ensures the default behavior of API remains unchanged and is backward compatible. If the confusion is arising from the choice of names of params, would ```exclude_mapping_stats``` and ```exclude_analysis_stats``` be a better logical choice?

Reply:
@mgodwan I would prefer the uri based approach. But, like @SwethaGuptha mentioned we need to maintain backward compatibility so can't turn it off by default. Now, we can explore the URI approach where metric can excluded/ included in via sub-metric